### DISCUSSION TITLE
Depth image to world coordinates

### DISCUSSION THREAD

**Discussion by smfloery** [pyvista/pyvista#2591] [Q&A]
Dear all,
I'm using pyvistagl to render a depth image using "get_image_depth()". Now I want to calculate the world coordinates for each pixel based on the camera and the depth image. For VTK I could find vtkDepthImageToPointCloud (https://vtk.org/doc/nightly/html/classvtkDepthImageToPointCloud.html#details). I haven't found something similar in pyvista. 

Given that one can extract the projection and the view matrix from the camera, it should be straightforward to calculate the world coordinates. But unfortunately I cant figure out how to extract the respective matrices from the pyvista camera? The only matrix which is available as attribute is the "model_transform_matrix".

Thank you very much
Sebastian

**Comment by MatthewFlamm** [comment]
I'm not very familiar with your use case, but is something like this what you need?

https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.camera_position.html

https://docs.pyvista.org/api/core/_autosummary/pyvista.Camera.position.html#pyvista.Camera.position

**Reply by smfloery** [reply]
Dear Mathew,
thanks for your answer. Probably I need to specify my question: I'm not looking for the world coordinates of the camera itself (which I could get with the functions you mentioned). Given that I have a rendered scene for which I obtained the depth image I want to calculate the world coordinates of each pixel.
